app/policies/stock_policy.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+class StockPolicy < ApplicationPolicy
+  def index?
+    user.present?
+  end
+
+  def show?
+    teacher_or_admin_required? || (student_required? && !record.archived?)
```

Comment:
```suggestion
    user.present?
```
All viewers can see the index and show pages for stocks.